addon/mixins/in-viewport.js

```diff
@@ -139,14 +141,16 @@ export default Mixin.create({
    * @param {Array} - entries
    */
   _onIntersection(entries) {
-    const entry = entries[0];
-
-    if (entry.isIntersecting) {
-      set(this, 'viewportEntered', true);
-      this.trigger('didEnterViewport');
-    } else if (entry.intersectionRatio <= 0) { // exiting viewport
-      set(this, 'viewportEntered', false);
-      this.trigger('didExitViewport');
+    if (!this.isDestroyed && !this.isDestroying) {
```

Comment:
This is a good catch!  I’m guessing you saw this error when trying out this branch?

Reply:
I see this error now.  Nice